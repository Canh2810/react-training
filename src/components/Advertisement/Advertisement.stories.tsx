import { StoryObj, Meta } from '@storybook/react'

import Advertisement from '.'

export default {
  title: 'Components/Advertisement',
  component: Advertisement,
} as Meta

type Story = StoryObj<typeof Advertisement>

export const Default: Story = {}
