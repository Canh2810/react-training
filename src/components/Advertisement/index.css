.advertisement {
  background-color: var(--background-tertiary-color);
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 750px;
  height: 100px;
  margin: 0 auto;
}
