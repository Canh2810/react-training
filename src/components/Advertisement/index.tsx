// Types
import { TypoColors, TypoVariants } from '@types'

// Components
import { Typography } from '../'

// Style
import './index.css'

const Advertisement = () => (
  <div className="advertisement">
    <Typography variant={TypoVariants.TextExtraSmall}>Advertisement</Typography>
    <Typography
      variant={TypoVariants.HeadingMedium}
      color={TypoColors.Tertiary}
    >
      You can place ads
    </Typography>
    <Typography variant={TypoVariants.TextMedium}>750x100</Typography>
  </div>
)

export default Advertisement
