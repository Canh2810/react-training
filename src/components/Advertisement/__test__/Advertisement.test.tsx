import { render } from '@testing-library/react'
import Advertisement from '..'

describe('Advertisement component', () => {
  it('should render correctly', () => {
    const { container } = render(<Advertisement />)
    expect(container).toMatchSnapshot()
  })
})
