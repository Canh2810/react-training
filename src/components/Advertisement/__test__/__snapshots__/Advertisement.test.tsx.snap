// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advertisement component should render correctly 1`] = `
<div>
  <div
    class="advertisement"
  >
    <p
      class=""
      style="font-family: inter; font-size: 14px; font-weight: 400; line-height: 20px; color: rgb(59, 60, 74);"
    >
      Advertisement
    </p>
    <p
      class=""
      style="font-family: work-sans; font-size: 20px; font-weight: 600; line-height: 28px; color: rgb(105, 106, 117);"
    >
      You can place ads
    </p>
    <p
      class=""
      style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
    >
      750x100
    </p>
  </div>
</div>
`;
